src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+            self->widgets[widx::status_bar].bottom = self->height - 3;
+            self->widgets[widx::status_bar].right = self->width - 14;
+
+            self->widgets[widx::centre_on_viewport].right = self->widgets[widx::viewport].right - 1;
+            self->widgets[widx::centre_on_viewport].bottom = self->widgets[widx::viewport].bottom - 1;
+            self->widgets[widx::centre_on_viewport].left = self->widgets[widx::viewport].right - 24;
+            self->widgets[widx::centre_on_viewport].top = self->widgets[widx::viewport].bottom - 24;
+
+            common::repositionTabs(self);
+        }
+
+        // 0x0048E470
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+            self->drawViewports(dpi);
+            widget::drawViewportCentreButton(dpi, self, widx::centre_on_viewport);
+
+            auto station = stationmgr::get(self->number);
+            const char* buffer = stringmgr::get_string(string_ids::buffer_1250);
+            station->getStatusString((char*)buffer);
+
+            auto args = FormatArguments();
+            args.push(string_ids::buffer_1250);
+
+            const auto& widget = self->widgets[widx::status_bar];
+            const auto x = self->x + widget.left - 1;
+            const auto y = self->y + widget.top - 1;
+            const auto width = widget.width() - 1;
+            gfx::draw_string_494BBF(*dpi, x, y, width, colour::black, string_ids::white_stringid2, &args);
+        }
+
+        // 0x0048E4D4
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameStationPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_station:
+                case common::widx::tab_cargo:
+                case common::widx::tab_cargo_ratings:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                // 0x0049932D
+                case widx::centre_on_viewport:
+                {
+                    if (self->viewports[0] == nullptr || self->saved_view.isEmpty())
+                        break;
+
+                    registers regs;
+                    regs.ax = self->saved_view.mapX;
+                    regs.cx = self->saved_view.mapY & 0x3FFF;
+                    regs.dx = self->saved_view.surfaceZ;
+                    regs.esi = (int32_t)WindowManager::getMainWindow();
+                    call(0x004C6827, regs);
+                    break;
+                }
+            }
+        }
+
+        static void initViewport(window * self);
+
+        // 0x0048E70B
+        static void on_resize(window * self)
+        {
```

Comment:
I think you have purposely removed the sub_48E32C calls. What is the reasoning? It is used so that you can rename stations.

Reply:
This appears to be inspired by the line I left in the town window's resize events. There, it was initially nulled by Ted to allow renaming towns during game play.

I'm not sure this should apply to stations, too, though. I don't mind either way, but it should be noted in the changelog if we do.

Reply:
Yeah I got it from the townwnd.cpp. However on closer inspection the sub_48E32C prevents the player from renaming rival company stations. So I will include this.